osu.Game/Beatmaps/BeatmapStore.cs

```diff
@@ -32,6 +32,15 @@ public void Add(BeatmapSetInfo beatmapSet)
         {
             var context = GetContext();
 
+            foreach (var beatmap in beatmapSet.Beatmaps.Where(b => b.Metadata != null))

```

Comment:
i'm not sure we want to do this. we should be leaving the metadata field null where it inherits from its set, i think (so we can differentiate between set and beatmap metadata when edits occur).

Reply:
It is `null` though. The import makes sure of that.
This part just checks for duplicate metadata where metadata is present.
![qq](https://user-images.githubusercontent.com/27222621/33712552-256b71e6-db48-11e7-8071-a508ff18f1c3.png)

Reply:
So the `.Local` is checking yet-to-be-committed changes which is why this works? would definitely be worth adding a comment explaining what this is doing, as it's not immediately obvious.